src/OrleansServiceBus/Providers/EventDataGeneratorStreamProvider/NoOpCheckpointer.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Orleans.Streams;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.ServiceBus.Providers.Testing
+{
+    /// <summary>
+    /// NoOpCheckpointer is used in EventDataGeneratorStreamProvider eco system to replace the default Checkpointer which requires a back end storage. In EventHubDataGeneratorStreamProvider,
+    /// it is generating EventData on the fly when receiver pull messages from the queue, which means it doesn't support recoverable stream, hence check pointing won't bring much value there. 
+    /// So a checkpointer with no ops should be enough.
+    /// </summary>
+    public class NoOpCheckpointer : IStreamQueueCheckpointer<string>
+    {
+        public static Task<IStreamQueueCheckpointer<string>> Create()
+        {
+            return Task.FromResult<IStreamQueueCheckpointer<string>>(new NoOpCheckpointer());
+        }
+
+        public bool CheckpointExists => false;
```

Comment:
Should probably return true here, as false will trigger system to try to diligently create a checkpoint.